ray-curator/ray_curator/stages/filters/fasttext_filter.py

```diff
@@ -0,0 +1,92 @@
+# Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+
+import fasttext
+import numpy as np
+
+from ray_curator.stages.filters.doc_filter import DocumentFilter
+
+
+class FastTextQualityFilter(DocumentFilter):
+    def __init__(self, model_path: str | None = None, label: str = "__label__hq", alpha: float = 3, seed: int = 42):
+        if model_path is None:
+            msg = "Must provide a valid path to a FastText model to compute document scores with this filter"
+            raise ValueError(msg)
+        self._model_path = model_path
+        self._label = label
+        self._alpha = alpha
+        self._seed = np.random.seed(seed)  # noqa: NPY002
+        self._name = "fasttext_quality_filter"
+
+    def model_check_or_download(self) -> None:
+        if not os.path.exists(self._model_path):
+            msg = f"Model file {self._model_path} not found"
+            raise FileNotFoundError(msg)
+
+    def load_model(self) -> None:
+        self.model = fasttext.load_model(self._model_path)
+
+    def score_document(self, text: str) -> float:
+        # See setup() function in modules/filter.py
+        model = self.model
+
+        text = text.replace("\n", " ").replace("__label__", " ")
+        pred = model.predict(text)
```

Comment:
FastText's `predict` runs serially, so IMO adding the `@batched` decorator here would not help.